postgres/datadog_checks/postgres/postgres.py

```diff
@@ -120,62 +131,12 @@ def _get_replication_role(self):
         # value fetched for role is of <type 'bool'>
         return "standby" if role else "master"
 
-    def _get_version(self):
-        if self.version is None:
-            cursor = self.db.cursor()
-            cursor.execute('SHOW SERVER_VERSION;')
-            version = cursor.fetchone()[0]
-            try:
-                version_parts = version.split(' ')[0].split('.')
-                version = [int(part) for part in version_parts]
-            except Exception:
-                # Postgres might be in development, with format \d+[beta|rc]\d+
-                match = re.match(r'(\d+)([a-zA-Z]+)(\d+)', version)
-                if match:
-                    version_parts = list(match.groups())
-
-                    # We found a valid development version
-                    if len(version_parts) == 3:
-                        # Replace development tag with a negative number to properly compare versions
-                        version_parts[1] = -1
-                        version = [int(part) for part in version_parts]
-
-            self.version = version
-
-        self.service_metadata('version', self.version)
-        return self.version
-
-    def _is_above(self, version_to_compare):
-        version = self._get_version()
-        if type(version) == list:
-            # iterate from major down to bugfix
-            for v, vc in zip_longest(version, version_to_compare, fillvalue=0):
-                if v == vc:
-                    continue
-
-                return v > vc
-
-            # return True if version is the same
-            return True
-        return False
-
-    def _is_8_3_or_above(self):
-        return self._is_above([8, 3, 0])
-
-    def _is_9_1_or_above(self):
-        return self._is_above([9, 1, 0])
-
-    def _is_9_2_or_above(self):
-        return self._is_above([9, 2, 0])
-
-    def _is_9_4_or_above(self):
-        return self._is_above([9, 4, 0])
-
-    def _is_9_6_or_above(self):
-        return self._is_above([9, 6, 0])
-
-    def _is_10_or_above(self):
-        return self._is_above([10, 0, 0])
+    @property
+    def version(self):
```

Comment:
We shouldn't cache version information as it may change

Reply:
It's only cleared when an error occurs and the difference between minor releases in unlikely to cause one

Reply:
Ok, that is right and that would affect inventories project only. So I'll remove the caching as part of the PR that adds inventories. This PR is not intending to change behaviour and version was already cached (https://github.com/DataDog/integrations-core/pull/4874#issuecomment-547456682)